app/controllers/posts_controller.rb

```diff
@@ -11,4 +11,21 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new, alert: 'Failed to create post.'
+    end
+  end
```

Comment:
- [ ] According to the requirements, a post should be created on behalf of the currently logged in user (`current_user`) of the app. You are currently setting the current user based on the `user_id` params. Please make this change 👍